java/vector/src/main/java/org/apache/arrow/vector/dictionary/DictionaryEncoder.java

```diff
@@ -98,10 +98,11 @@ public static ValueVector decode(ValueVector indices, Dictionary dictionary) {
     // copy the dictionary values into the decoded vector
     TransferPair transfer = dictionaryVector.getTransferPair(indices.getAllocator());
     transfer.getTo().allocateNewSafe();
+
+    BaseIntVector baseIntVector = (BaseIntVector) indices;
     for (int i = 0; i < count; i++) {
-      Object index = indices.getObject(i);
-      if (index != null) {
-        int indexAsInt = ((Number) index).intValue();
+      if (!baseIntVector.isNull(i)) {
```

Comment:
encoded value can never be null?

Reply:
Nope, see https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/dictionary/DictionaryEncoder.java#L72

Reply:
I see. The benefit of not encoding null is efficient storage. However, the benefit of encoding null is efficient computation (no if branch in the encoding/decoding logic, which makes the CPU pipeline/prefetch more efficient). 

We may need some investigation on this. For this particular case, it may be OK to stick to the original assumption.